packages/esm-patient-orders-app/src/api/api.ts

```diff
@@ -67,13 +74,27 @@ export function createEmptyEncounter(
     );
     visitStartDate;
   }
+
+  const { items, postDataPrepFunctions }: OrderBasketStore = orderBasketStore.getState();
+  const patientItems = items[patientUuid];
+
+  const orders: Array<OrderPost> = [];
+  for (let grouping in patientItems) {
+    const groupOrders = patientItems[grouping];
+    for (let i = 0; i < groupOrders.length; i++) {
+      const order = groupOrders[i];
+      orders.push(postDataPrepFunctions[grouping](order, patientUuid, null));
+    }
+  }
```

Comment:
Instead of getting the `orders` from the store, the orders should be passed via the function arguments.

Reply:
Also, the function `createEmptyEncounter` should be renamed to `saveOrdersWithNewEncounter`, since we are also saving the orders in the same encounter request.